### PR TITLE
Bulk Data Import with missing 'id' on Audit #3016

### PR DESCRIPTION
**Pull request by prb112**
- Switched the conditionalUpdate to skip and pass in the oldResource as
- they exactly match rather than the Meta updated version with the unused new version
- update the Create to ignore the Version when passed in
- Enable the Version on Update by using the Create with Version.
- Refactor the BulkAuditLogger to eliminate the dual resource usage

Signed-off-by: Paul Bastide <pbastide@us.ibm.com>

### CODE REVIEW

File: fhir-bulkdata-webapp/src/main/java/com/ibm/fhir/bulkdata/audit/BulkAuditLogger.java [hunk 1]
```diff
@@ -60,15 +60,15 @@ public boolean shouldLog() {
         return svc.isEnabled();
```
Comment (tbieste) [inline fhir-bulkdata-webapp/src/main/java/com/ibm/fhir/bulkdata/audit/BulkAuditLogger.java:60]: Update copyright date.

**Review by tbieste — APPROVED** [review]

**Comment by lmsurpre** [comment]
Only tangentially related to this PR but I'm wondering whether https://github.com/IBM/FHIR/pull/3314 will have an impact on the audit log message for our delete interactions

**Comment by prb112** [comment]
> Only tangentially related to this PR but I'm wondering whether #3314 will have an impact on the audit log message for our delete interactions

It probably should

**Review by lmsurpre — COMMENTED** [review]
Will change our audit log payload?
Probably fine either way, but wondering if we need to update our docs at all...like `attachments/content/data/resource_type` under https://ibm.github.io/FHIR/guides/FHIRServerUsersGuide#4111-cadf-audit-log-entry

**Comment by prb112** [comment]
that's a good question, seems like an issue to confirm.

**Pull request merged**